lib/spack/spack/environment.py

```diff
@@ -707,6 +708,17 @@ def _read_manifest(self, f, raw_yaml=None):
         configuration = config_dict(self.yaml)
         self.concretization = configuration.get('concretization')
 
+        # Retrieve dev-build packages:
+        self.dev_specs = configuration['develop']
+        for name, entry in self.dev_specs.items():
+            # spec must include a concrete version
+            assert Spec(entry['spec']).version.concrete
+            # do not check paths because `spack develop` will clone all develop
+            # specs when run with no args
```

Comment:
Because we're validating one of the two fields in a dictionary -- why aren't we validating the other one?

I can remove it if it's just confusing the issue, but that was the thought.

Reply:
> Because we're validating one of the two fields in a dictionary -- why aren't we validating the other one?

OK good point - apologies but maybe I'm confused in the other direction now: couldn't we validate that

* the path is set (it doesn't look like `None` or `''` would be acceptable looking at the `spack develop` logic
* the path is a valid path (to be clear, I'm not saying whether the path exists or not, since `spack develop` can create the path if it doesn't exist, but rather just make sure we don't have a string like `"----"` (which isn't a valid filename)

?

Reply:
We could do some validation on it of that sort. I'll figure out a regex for filenames to check it against.